dspace-server-webapp/src/main/java/org/dspace/app/rest/PrimaryBitstreamController.java

```diff
@@ -0,0 +1,161 @@
+import org.dspace.app.rest.exception.UnprocessableEntityException;
+import org.dspace.app.rest.model.BitstreamRest;
+import org.dspace.app.rest.model.BundleRest;
+import org.dspace.app.rest.model.hateoas.BundleResource;
+import org.dspace.app.rest.repository.BundlePrimaryBitstreamLinkRepository;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.Bitstream;
+import org.dspace.content.DSpaceObject;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ControllerUtils;
+import org.springframework.hateoas.RepresentationModel;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This RestController is responsible for managing primaryBitstreams on bundles.
+ * The endpoint can be found at /api/core/bundles/{bundle-uuid}/primaryBitstream
+ */
+@RestController
+@RequestMapping("/api/" + BundleRest.CATEGORY + "/" + BundleRest.PLURAL_NAME
+    + REGEX_REQUESTMAPPING_IDENTIFIER_AS_UUID + "/" + BundleRest.PRIMARY_BITSTREAM)
+public class PrimaryBitstreamController {
+
+    @Autowired
+    private BundlePrimaryBitstreamLinkRepository repository;
+    @Autowired
+    private ConverterService converter;
+    @Autowired
+    private Utils utils;
+
+    /**
+     * This method retrieves a primaryBitstream on the given Bundle.
+     * Returns null if Bundle doesn't have a primaryBitstream.
+     * <br><code>
+     * curl -X GET "http://{dspace.server.url}/api/core/bundles/{bundle-uuid}/primaryBitstream"
+     * </code>
+     *
+     * @param uuid      The UUID of the Bundle of which the primaryBitstream will be retrieved
+     * @param request   The HttpServletRequest
+     * @return          The primaryBitstream, or null if not found
+     */
+    @PreAuthorize("hasPermission(#uuid, 'BUNDLE', 'READ')")
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<RepresentationModel<?>> getPrimaryBitstream(@PathVariable UUID uuid,
+                                                 HttpServletRequest request) {
+        BitstreamRest bitstreamRest = repository.getPrimaryBitstream(null, uuid, null, utils.obtainProjection());
+        if (bitstreamRest == null) {
+            return ControllerUtils.toEmptyResponse(HttpStatus.NO_CONTENT);
+        } else {
+            return ControllerUtils.toResponseEntity(HttpStatus.OK, new HttpHeaders(),
+                                                    (RepresentationModel<?>) converter.toResource(bitstreamRest));
+        }
+    }
```

Comment:
is this method really needed? it is a duplicate of https://github.com/DSpace/DSpace/pull/8876/files#diff-82de9d17a10a154ed57fc9caee29a588285e5676d070bf2917d2340775840cd8R50
I guess that you need this method because this controller will catch all the request to the `/primaryBitstream` path but you can eventually try to restrict the controller to the POST and PUT method and see if spring would fallback GET request to our standard controller.
The LinkRepository is probably needed when we embedd the primaryBitstream in the bundle